src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1452,9 +1450,9 @@ TEST(SpatialPoolerTest, testRaisePermanencesToThreshold) {
         potential.push_back(j);
       }
     }
-    UInt connected = sp.raisePermanencesToThreshold_(perm, potential);
-    ASSERT_TRUE(check_vector_eq(truePerm[i], perm));
-    ASSERT_TRUE(connected == trueConnectedCount[i]);
+    sp.raisePermanencesToThreshold_(perm, potential);
+    for( UInt syn = 0; syn < perm.size(); syn++ )
+      ASSERT_NEAR( truePerm[i][syn], perm[syn], synPermBelowStimulusInc );
```

Comment:
Nice change in test 👍
But the allowed diff is too large

Reply:
Using a maximum abs-diff of `synPermBelowStimulusInc` makes the test work, because the rewritten SP method ignores `synPermBelowStimulusInc` and so it is the largest discrepancy between the original results (which the test case looks for) and the SP's new results.

Also, I just now added to the test case to ensure that regardless of exact values, enough synapses are connected.